nxos_routing.tf

```diff
@@ -0,0 +1,214 @@
+locals {
+  routing_ipv4_prefix_list_rule = flatten([
+    for device in local.devices : [
+      for ipv4_prefix_list in try(local.device_config[device.name].routing.ipv4_prefix_lists, []) : {
+        key    = format("%s/%s", device.name, ipv4_prefix_list.name),
+        device = device.name,
+        name   = ipv4_prefix_list.name
+      }
+    ]
+  ])
+}
+
+resource "nxos_ipv4_prefix_list_rule" "ipv4_prefix_list_rule" {
+  for_each = { for v in local.routing_ipv4_prefix_list_rule : v.key => v }
+  device   = each.value.device
+  name     = each.value.name
+}
+
+locals {
+  routing_ipv4_prefix_list_rule_entry = flatten([
+    for device in local.devices : [
+      for ipv4_prefix_list in try(local.device_config[device.name].routing.ipv4_prefix_lists, []) : [
+        for entry in try(ipv4_prefix_list.entries, []) : {
+          key        = format("%s/%s/%s", device.name, ipv4_prefix_list.name, entry.order),
+          device     = device.name,
+          rule_name  = ipv4_prefix_list.name,
+          order      = entry.order,
+          action     = try(entry.action, local.defaults.nxos.configuration.routing.ipv4_prefix_lists.entries.action, "permit"),
+          prefix     = try(entry.prefix, null),
```

Comment:
For anything that does not have a uniqueness constraint (like the prefix list name) it's best to apply the following pattern:

```hcl
prefix = try(entry.prefix, local.defaults.nxos.configuration.routing.ipv4_prefix_lists.entries.action, null)
```

Even if we don't provide a default value as part of the module, it still allows the user to configure its own default values.